src/typescript/frontend/src/components/pages/metadata/MetadataPage.tsx

```diff
@@ -0,0 +1,276 @@
+            return 1;
+          }
+          return a[0].localeCompare(b[0]);
+        })
+        .map(([key, value]) => (
+          <FormEntry
+            key={key}
+            name={key}
+            value={value}
+            placeholder={PLACEHOLDERS.get(key) ?? key}
+            onChange={(e) => setFields(new Map(fields.set(key, e.target.value)))}
+            onClick={() => {
+              fields.delete(key);
+              setFields(new Map(fields));
+            }}
+            buttonText="-"
+          ></FormEntry>
+        ))}
+      <FormEntry
+        name="Add new field"
+        value={addFieldName}
+        placeholder="New field name"
+        onChange={(e) => {
+          setAddFieldName(e.target.value);
+        }}
+        onClick={() => {
+          if (addFieldName === "") {
+            return;
+          }
+          setFields(new Map(fields.set(addFieldName, "")));
+          setAddFieldName("");
+        }}
+        disabled={addFieldName === ""}
+        buttonText="+"
+      ></FormEntry>
+      <div>{/* For spacing. */}</div>
+      <div className="mx-auto w-[300px] grid grid-cols-2 grid-rows-2 gap-[10px]">
+        <button
+          tabIndex={3}
+          className={`${BUTTON_CLASSNAME}`}
+          onClick={() => {
+            const data = {
+              marketAddress,
+              fields: Array.from(fields.entries().filter(([_, value]) => value !== "")),
+            };
+            navigator.clipboard.writeText(JSON.stringify(data));
+          }}
+        >
+          Copy
+        </button>
+        <button
+          tabIndex={3}
+          className={`${BUTTON_CLASSNAME}`}
+          onClick={() => {
+            navigator.clipboard
+              .readText()
+              .then((r) => {
+                const data = JSON.parse(r);
+                setFields(new Map(data.fields));
+                setMarketAddress(data.marketAddress);
+                setPasted(true);
+              })
+              .catch(console.warn);
+          }}
+        >
+          Paste
+        </button>
+        <button
+          tabIndex={3}
+          className={`${BUTTON_CLASSNAME} ${!isSubmitEnabled ? "!border-dark-gray" : ""} col-span-full`}
+          onClick={async () => {
+            if (!isSubmitEnabled) {
+              return;
+            }
+            const builderLambda = () =>
+              SetMarketProperties.builder({
+                aptosConfig: aptos.config,
+                admin: account!.address,
+                market: marketAddress,
+                keys: Array.from(
+                  fields
+                    .entries()
+                    .filter(([_, value]) => value !== "")
+                    .map(([key, _]) => key)
+                ),
+                values: Array.from(
+                  fields
+                    .entries()
+                    .filter(([_, value]) => value !== "")
+                    .map(([_, value]) => value)
+                ),
```

Comment:
Can probably clean this up a bit, not a big deal though

Reply:
Fixed in f8a7353b87d46bad1fbfa010e10d2f0ab9ad9176.